File: 1848.minimum-distance-to-the-target-element.js
/*
 * @lc app=leetcode id=1848 lang=javascript
 *
 * [1848] Minimum Distance to the Target Element
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @param {number} start
 * @return {number}
 */
var getMinDistance = function (nums, target, start) {
    let res = nums.length;
    for (let i in nums) {
        if (i >= start + res) {
            break;
        }
        if (nums[i] == target) {
            res = Math.min(res, Math.abs(start - i));
        }
    }
    return res;
};
// @lc code=end

File: cpp/1848.minimum-distance-to-the-target-element.cpp
/*
 * @lc app=leetcode id=1848 lang=cpp
 *
 * [1848] Minimum Distance to the Target Element
 */
#include <vector>
#include <algorithm>
using namespace std;
// @lc code=start
class Solution
{
public:
    int getMinDistance(vector<int> &nums, int target, int start)
    {
        int width = nums.size();
        int res = width;
        for (int i = 0; i < min(width, start + res); ++i)
        {
            if (nums[i] == target)
            {
                res = min(res, abs(i - start));
            }
        }
        return res;
    }
};
// @lc code=end
